runtime/convert.c

```diff
@@ -14,3 +15,33 @@ TaggedPtr _bal_convert_to_float(TaggedPtr tp) {
     }
     return floatToTagged(d);
 }
+
+TaggedWithOverflow _bal_convert_to_int(TaggedPtr tp) {
+    int64_t val;
+    TaggedWithOverflow res;
+    switch (getTag(tp) & UT_MASK) {
+        case TAG_FLOAT:
+            {
+                IntWithOverflow intRes = _bal_float_to_int(taggedToFloat(tp));
```

Comment:
Declare intRes just before res. Then get rid of the braces in the cases. Assign res.overflow once. Get rid of val.

Reply:
Changed